### PR TITLE
Remove some stubbing

### PR DESCRIPTION
**Pull request by jhawthorn**
Removed some stubbing in specs in favour of using real objects.

Didn't remove all of the stubs in these files, but this should be in the right direction.

### CODE REVIEW

**Comment by magnusvk** [comment]
:+1: Spec failure seemed unrelated, yeah?

**Comment by jhawthorn** [comment]
Yes. Opened #280

**Comment by jordan-brough** [comment]
sweet. :+1:

**Comment by gmacdougall** [comment]
:+1:

File: core/spec/models/spree/order_updater_spec.rb [hunk 1]
```diff
@@ -238,33 +232,23 @@ module Spree
...
-        shipments = [shipment]
-        allow(order).to receive_messages shipments: shipments
-        allow(shipments).to receive_messages states: []
-        allow(shipments).to receive_messages ready: []
-        allow(shipments).to receive_messages pending: []
-        allow(shipments).to receive_messages shipped: []
+        create(:shipment, order: order)
 
-        expect(shipment).to receive(:update!).with(order)
+        expect_any_instance_of(Spree::Shipment).to receive(:update!)
```
Comment (athal7) [inline core/spec/models/spree/order_updater_spec.rb]: i don't think this really tests what the example says it does anymore. the example says that each shipment gets updated, but this just asserts than any shipment somewhere gets updated. same for the specs below

**Comment by jhawthorn** [comment]
@athal7 reverted that portion back to `expect(shipment).to receive(...)`

Would be nice to eventually test the behaviour rather than what is called, but I did not get that far in this PR.

**Comment by athal7** [comment]
:+1:

**Pull request merged**